Author: Dominik George <nik@naturalnet.de>
Description: Adds an option --onemaildir when fetching feeds
 Place all articles directly into mailfolder. Normally, rssdrop will 
 create sub directories for each feed. If this option is used when
 fetching a feed, the fetched articles are written to the mailfolder
 directly instead of a subdirectory.
Forwarded: no
--- a/rssdrop
+++ b/rssdrop
@@ -20,6 +20,7 @@ my %opt = (
 my @getopt =
   qw/prefix=s
      mailfolder=s
+     onemaildir
      add
      delete
      fetch!
@@ -109,6 +110,7 @@ sub usage {
     --umask NUM                specify octal umask for new files [077]
     --prefix DIR               path/to/rssdrop/dir
     --mailfolder DIR           path/to/maildir/folder/area
+    --onemaildir               place all articles directly in mailfolder
     --verbose
     --help
   };
@@ -153,7 +155,7 @@ sub load_feed {
   $feed->{name} = $feedname;
   $feed->{folder} ||= $feedname;
   $feed->{dir} = "$opt{prefix}/feeds/$feed->{folder}";
-  $feed->{maildir} = "$config->{mailfolder}/$feed->{folder}";
+  $feed->{maildir} = $opt{onemaildir} && "$config->{mailfolder}" || "$config->{mailfolder}/$feed->{folder}";
 
   init_feeddir($feed->{dir});
   init_maildir($feed->{maildir});
@@ -707,6 +709,13 @@ setup, however, rssdrop saves your $mail
 section of the config file. If you're happy with this setting, there's
 no need to specify --mailfolder after that.
 
+=item --onemaildir
+
+Place all articles directly into mailfolder. Normally, rssdrop will
+create sub directories for each feed. If this option is used when
+fetching a feed, the fetched articles are written to the mailfolder
+directly instead of a subdirectory.
+
 =item --verbose 
 
 Turn on extra logging messages. By default, rssdrop is quiet.
